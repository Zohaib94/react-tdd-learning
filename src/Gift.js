import React, { Component } from 'react';

export default class Gift extends Component {
  state = {
    person: '',
    present: ''
  }

  render() {
    return (
      <div>
        <form>
          <label>Person:</label>
          <input type='text' className='person-input' onChange={(event) => {
            this.setState({
              person: event.target.value
            })
          }} />

          <label>Present:</label>
          <input type='text' className='present-input' onChange={(event) => {
            this.setState({
              present: event.target.value
            })
          }} />
        </form>
      </div>
    )
  }
}
