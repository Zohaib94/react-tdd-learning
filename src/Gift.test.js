import React from 'react'
import { shallow } from 'enzyme'
import Gift from './Gift'

describe('Gift', () => {
  const gift = shallow(<Gift/>);

  it('renders properly', () => {
    expect(gift).toMatchSnapshot();
  })

  it('is expected that initial state is empty', () => {
    expect(gift.state()).toEqual({
      person: '',
      present: ''
    })
  })

  describe('When person input is entered', () => {
    const personInput = 'Zohaib'

    beforeEach(() => {
      gift.find('.person-input').simulate('change', { target: { value: personInput }})
    })

    it('is expected that state is equivalent to person input', () => {
      expect(gift.state().person).toEqual(personInput)
    })
  })

  describe('When present input is entered', () => {
    const presentInput = 'Zohaib'

    beforeEach(() => {
      gift.find('.present-input').simulate('change', { target: { value: presentInput }})
    })

    it('is expected that state is equivalent to present input', () => {
      expect(gift.state().present).toEqual(presentInput)
    })
  })
})
