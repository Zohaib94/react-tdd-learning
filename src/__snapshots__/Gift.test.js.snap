// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
<div>
  <form>
    <label>
      Person:
    </label>
    <input
      className="person-input"
      onChange={[Function]}
      type="text"
    />
    <label>
      Present:
    </label>
    <input
      className="present-input"
      onChange={[Function]}
      type="text"
    />
  </form>
</div>
`;
